code/game/objects/items/weapons/tools.dm

```diff
@@ -305,66 +295,75 @@
+/obj/item/weldingtool/attack(mob/living/M, mob/user, var/target_zone)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		var/obj/item/organ/external/S = H.organs_by_name[target_zone]
 
-		var/obj/item/organ/external/S = M:organs_by_name[target_zone]
+		if(!S)
+			return
 
-		if (!S) return
 		if(!(S.status & ORGAN_ASSISTED) || user.a_intent != I_HELP)
 			return ..()
 
-		if(M.isSynthetic() && M == user && !(M.get_species() == "Military Frame"))
+		if(H.isSynthetic() && H == user && !(H.get_species() == "Military Frame"))
 			to_chat(user, "<span class='warning'>You can't repair damage to your own body - it's against OH&S.</span>")
 			return
-		if(S.brute_dam == 0)
-			// Organ undamaged
-			to_chat(user, "Nothing to fix here!")
-			return
-		if (!src.welding)
-			// Welder is switched off!
-			to_chat(user, "<span class='warning'>You need to light the welding tool, first!</span>")
+
+		if (!welding)
+			to_chat(user, "<span class='warning'>You need to light the welding tool first!</span>")
 			return
 
 		if(S.brute_dam > ROBOLIMB_SELF_REPAIR_CAP && (S.status & ORGAN_ROBOT))
-			user << "<span class='warning'>The damage is far too severe to patch over externally.</span>"
+			to_chat(user, "<span class='warning'>The damage is far too severe to patch over externally!</span>")
 			return
 
-		if (src.remove_fuel(0))
-			// Use a bit of fuel and repair
-			S.heal_damage(15,0,0,1)
-			user.setClickCooldown(DEFAULT_ATTACK_COOLDOWN)
-			user.visible_message("<span class='warning'>\The [user] patches some dents on \the [M]'s [S.name] with \the [src].</span>")
-		else
-			// Welding tool is out of fuel
-			to_chat(user, "Need more welding fuel!")
-		return
+		repair_organ(user, H, S)
 
 	else
 		return ..()
 
-/obj/item/weldingtool/afterattack(obj/O as obj, mob/user as mob, proximity)
-	if(!proximity) return
-	if (istype(O, /obj/structure/reagent_dispensers/fueltank) && get_dist(src,O) <= 1 && !src.welding)
+/obj/item/weldingtool/proc/repair_organ(var/mob/living/user, var/mob/living/carbon/human/target, var/obj/item/organ/external/affecting)
+	if(!affecting.brute_dam)
+		user.visible_message("<span class='notice'>[user] finishes repairing the physical damage on [target]'s [affecting.name].</span>")
+		return
+
+	if(do_mob(user, target, 30))
+		if(remove_fuel(0))
+			var/static/list/repair_messages = list(
+				"patches some dents",
+				"mends some tears",
+				"repairs some joints"
+			)
+			affecting.heal_damage(brute = 15, robo_repair = TRUE)
+			user.visible_message("<span class='notice'>\The [user] [pick(repair_messages)] on [target]'s [affecting.name] with \the [src].</span>")
+			playsound(target, 'sound/items/Welder2.ogg', 15)
+			repair_organ(user, target, affecting)
+		else
+			to_chat(user, "<span class='warning'>\The [src] flickers off!</span>")
+
+/obj/item/weldingtool/afterattack(obj/O, mob/user, proximity)
+	if(!proximity)
+		return
+	if(istype(O, /obj/structure/reagent_dispensers/fueltank) && get_dist(src, O) <= 1 && !welding)
```

Comment:
```suggestion
	if(istype(O, /obj/structure/reagent_dispensers/fueltank) && Adjacent(O) && !welding)
```

I think Adjacent is better to use here because it does all the 'ClickCross' stuff.

Also elsewhere.